src/ros/ros2/ros2.ts

```diff
@@ -29,10 +29,27 @@ export class ROS2 implements ros.ROSApi {
         }));
     }
 
-    public getPackages(): Promise<{ [name: string]: string }> {
-        return new Promise((resolve, reject) => {
-            reject("not yet implemented");
-        });
+    public async getPackages(): Promise<{ [name: string]: () => Promise<string> }> {
+        let packages: { [name: string]: () => Promise<string> } = {};
+        const {stdout} = child_process.exec("ros2 pkg list", { env: this._env });
+        let chucks = "";
+        for await (const chuck of stdout) {
+            chucks += chuck;
+        }
+
+        chucks.split(os.EOL).map(((line) => {
+            const packageName:string = line.trim();
```

Comment:
nit: `const packageName: string`, we could also fix all errors like this later when we add linter check